pkg/apis/v1alpha1/dnshealthcheckprobe_types.go

```diff
@@ -0,0 +1,83 @@
+/*
+Copyright 2023 The MultiCluster Traffic Controller Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// DNSHealthCheckProbeSpec defines the desired state of DNSHealthCheckProbe
+type DNSHealthCheckProbeSpec struct {
+	Port              int               `json:"port,omitempty"`
+	Host              string            `json:"host,omitempty"`
+	Address           string            `json:"address,omitempty"`
+	Path              string            `json:"path,omitempty"`
+	Protocol          HealthProtocol    `json:"protocol,omitempty"`
+	Interval          metav1.Duration   `json:"interval,omitempty"`
+	AdditionalHeaders AdditionalHeaders `json:"additionalHeaders,omitempty"`
+	FailureThreshold  *int              `json:"failureThreshold,omitempty"`
+	ExpectedReponses  []int             `json:"ExpectedReponses,omitempty"`
```

Comment:
```suggestion
	ExpectedReponses  []int             `json:"expectedReponses,omitempty"`
```

Typo in the `json` flag

Reply:
nice, thanks. Fixed now.